examples/Analytic_300hPa_Trough.py

```diff
@@ -0,0 +1,179 @@
+                + 200 * np.cos(y * np.pi) + 300 * y * np.cos(x * np.pi + np.pi / 2))
+    elif parameter == 'temp':
+        return (-50 + 20 * np.cos(p * x * np.pi) * np.cos(q * y * np.pi) / 10
+                + 20 * np.cos(y * np.pi) / 10 + 5 * y * np.cos(x * np.pi + np.pi / 2) / 10)
+
+
+def lifting_300hPa_trough(parameter='hght'):
+    """ Lifting trough with heights and Temperatures based on Sanders Analytic Model
+    """
+    X = np.linspace(.25, .75, 101)
+    Y = np.linspace(.25, .75, 101)
+
+    x, y = np.meshgrid(X, Y)
+
+    p = 4
+    q = 2
+
+    if parameter == 'hght':
+        return (9240 + 150 * np.cos(p * x * np.pi) * np.cos(q * y * np.pi)
+                + 200 * np.cos(y * np.pi) + 400 * y * np.cos(x * np.pi + np.pi))
+    elif parameter == 'temp':
+        return (-50 + 20 * np.cos(p * x * np.pi) * np.cos(q * y * np.pi) / 10
+                + 20 * np.cos(y * np.pi) / 10 + 50 * y * np.cos(x * np.pi + np.pi) / 10)
+
+
+def digging_300hPa_trough(parameter='hght'):
+    """ Digging trough with heights and Temperatures based on Sanders Analytic Model
+    """
+    X = np.linspace(.25, .75, 101)
+    Y = np.linspace(.25, .75, 101)
+
+    x, y = np.meshgrid(X, Y)
+
+    p = 4
+    q = 2
+
+    if parameter == 'hght':
+        return (9240 + 150 * np.cos(p * x * np.pi) * np.cos(q * y * np.pi)
+                + 200 * np.cos(y * np.pi) + 400 * y * np.sin(x * np.pi + 5 * np.pi / 2))
+    elif parameter == 'temp':
+        return (-50 + 20 * np.cos(p * x * np.pi) * np.cos(q * y * np.pi) / 10
+                + 20 * np.cos(y * np.pi) / 10 + 50 * y * np.sin(x * np.pi + np.pi / 2) / 10)
+
+
+######################################################################
+# Call the appropriate definition to develop the desired wave.
+#
+
+# Single Trough
+Z = single_300hPa_trough(parameter='hght')
+T = single_300hPa_trough(parameter='temp')
+
+# Lifting Trough
+# Z = lifting_300hPa_trough(parameter='hght')
+# T = lifting_300hPa_trough(parameter='temp')
+
+# Digging Trough
+# Z = digging_300hPa_trough(parameter='hght')
+# T = digging_300hPa_trough(parameter='temp')
+
+
+######################################################################
+# Set geographic parameters for analytic grid to then
+#
+
+lats = np.linspace(35, 50, 101)
+lons = np.linspace(260, 290, 101)
+lon, lat = np.meshgrid(lons, lats)
+
+# Calculate Geostrophic Wind from Analytic Heights
+f = mpcalc.coriolis_parameter(lat * units('degrees'))
+dx, dy = mpcalc.lat_lon_grid_deltas(lons, lats)
+ugeo, vgeo = mpcalc.geostrophic_wind(Z*units.meter, f, dx, dy, dim_order='yx')
+
+# Get the wind direction for each point
+wdir = mpcalc.wind_direction(ugeo, vgeo)
+
+# Compute the Gradient Wind via an approximation
+dydx = mpcalc.first_derivative(Z, delta=dx, axis=1)
+d2ydx2 = mpcalc.first_derivative(dydx, delta=dx, axis=1)
+R = ((1+dydx.m**2)**(3./2.))/d2ydx2.m
+
+geo_mag = mpcalc.wind_speed(ugeo, vgeo)
+grad_mag = geo_mag.m - (geo_mag.m**2)/(f.magnitude*R)
```

Comment:
Can we space out some of the operators in the next two lines as well?

Reply:
Yes.